src/shared/efivars.c

```diff
@@ -193,6 +193,17 @@ int efi_set_reboot_to_firmware(bool value) {
         return 0;
 }
 
+char* efi_variable_path(sd_id128_t vendor, const char *name) {
+        _cleanup_free_ char *p = NULL;
```

Comment:
we don't need _cleanup_free_ here really... when asprintf() fails nothing is allocated hence there's nothing to free on failure

Reply:
Indeed. The man page says `p` is undefined, but a better word would be unset, looking at `__vasprintf_internal`. Glibc is careful to either not touch it at all (for negative return), or set it to a valid buffer. I'll adjust the code.